### PR TITLE
Add experimentalFeature flag for server side batch changes

### PR DESCRIPTION
**Pull request by eseliger**
.. so we can implement some UI bits without exposing them to users yet.

### CODE REVIEW

**Review by mrnugget — APPROVED** [review]

File: client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx [hunk 1]
```diff
@@ -53,11 +56,19 @@ const samples: Sample[] = [
...
+export interface CreateBatchChangePageProps extends SettingsCascadeProps<Settings> {
     headingElement: 'h1' | 'h2'
 }
 
-export const CreateBatchChangePage: React.FunctionComponent<CreateBatchChangePageProps> = ({ headingElement }) => {
+export const CreateBatchChangePage: React.FunctionComponent<CreateBatchChangePageProps> = ({
+    settingsCascade,
+    headingElement,
+}) => {
+    const isBatchChangesExecutionEnabled = Boolean(
```
Comment (courier-new) [inline client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx:67]: I think this is the first time I've ever seen `Boolean` used as a function like this! I had to go double-check the docs, but it's not any different from the `!!` operator, right? Do you prefer it for readability?

File: client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx [hunk 2]
```diff
@@ -53,11 +56,19 @@ const samples: Sample[] = [
...
+        settingsCascade !== null &&
+            !isErrorLike(settingsCascade.final) &&
+            settingsCascade.final?.experimentalFeatures?.batchChangesExecution
```
Comment (courier-new) [inline client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx:70]: First time seeing this settings cascade stuff for me, that's pretty cool!

File: client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx (hunk 1 above)
Comment (eseliger) [reply client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx:67]: Yes, it should be equivalent. I find them easier to understand, especially for people not so deep in the js universe. But I don't have a strong preference. In our codebase we have 87 vs 132 results for either of the two 😆

Comment (courier-new) [reply client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx:67]: Neat! Yeah no that makes a lot of sense. No strong preference myself, just not used to seeing it. 😄 Though the functional programmer part of me wishes it had a more obviously function-y name like `(expression).toBoolean()` or `parseBoolean(expression)` or something haha.

Comment (eseliger) [reply client/web/src/enterprise/batches/create/CreateBatchChangePage.tsx:67]: After all, this is still JAVAscript 😆

**Pull request merged**